cloud/amazon/lambda.py

```diff
@@ -0,0 +1,395 @@
+
+    # Update existing Lambda function
+    if state == 'present' and current_function:
+        current_config = current_function['Configuration']
+        current_code = current_function['Code']
+
+        # Update function configuration
+        func_kwargs = {'FunctionName': name}
+        if current_config['Role'] != role_arn:
+            func_kwargs.update({'Role': role_arn})
+        if current_config['Handler'] != handler:
+            func_kwargs.update({'Handler': handler})
+        if current_config['Description'] != description:
+            func_kwargs.update({'Description': description})
+        if current_config['Timeout'] != timeout:
+            func_kwargs.update({'Timeout': timeout})
+        if current_config['MemorySize'] != memory_size:
+            func_kwargs.update({'MemorySize': memory_size})
+
+        # Unsupported mutation
+        if not runtime:
+            module.fail_json(msg='runtime parameter is required')
+        if current_config['Runtime'] != runtime:
+            module.fail_json(msg='Cannot change runtime. Please recreate the function')
+
+        # Upload new configuration
+        if len(func_kwargs) > 1:
+            try:
+                if not check_mode:
+                    client.update_function_configuration(**func_kwargs)
+                changed = True
+            except (botocore.exceptions.ParamValidationError, botocore.exceptions.ClientError) as e:
+                module.fail_json(msg=str(e))
+
+        # Update code configuration
+        code_kwargs = {'FunctionName': name}
+
+        # Update S3 location
+        if s3_bucket and s3_key:
+            # If function is stored on S3 always update
+            code_kwargs.update({'S3Bucket': s3_bucket,
+                                'S3Key': s3_key})
+            if s3_object_version:
+                code_kwargs.update({'S3ObjectVersion': s3_object_version})
+
+        # Compare local checksum, update remote code when different
+        elif path:
+            local_checksum = sha256sum(path)
+            remote_checksum = current_config['CodeSha256']
+
+            # Only upload new code when local code is different compared to the remote code
+            if local_checksum != remote_checksum:
+                try:
+                    with open(path, 'rb') as f:
+                        encoded_zip = f.read()
+                    code_kwargs.update({'ZipFile' : encoded_zip})
+                except IOError as e:
+                    module.fail_json(msg=str(e))
+
+        # Upload new code
+        if len(code_kwargs) > 1:
+            try:
+                if not check_mode:
+                    response = client.update_function_code(**code_kwargs)
+                changed = True
+            except (botocore.exceptions.ParamValidationError, botocore.exceptions.ClientError) as e:
+                module.fail_json(msg=str(e))
+
+        # We're done
+        module.exit_json(changed=changed, result=response)
```

Comment:
If the function exists but there are no changes (thus `len(code_kwargs)` is just one item), this references a nonexistent response object & causes the task to fail.